bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,150 @@
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function count;
+use function is_numeric;
+use function sprintf;
+
+final class UpdateContentVisibilityCommand extends Command
+{
+    private SymfonyStyle $style;
+
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Updates content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+        $this->addOption(
+            'limit',
+            'l',
+            InputOption::VALUE_OPTIONAL,
+            'Number of content object to process in a single iteration',
+            50,
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->style->info(
+            'This command fetches content and updates visibility based on its schedule from publish_from and publish_to fields.',
+        );
+
+        if (!$this->configurationService->isEnabled()) {
+            $this->style->warning('Scheduled visibility mechanism is disabled.');
+
+            return Command::FAILURE;
+        }
+
+        $query = new Query();
+        $criteria = new Criterion\LogicalOr(
+            [
+                new Criterion\IsFieldEmpty('publish_from', false),
+                new Criterion\IsFieldEmpty('publish_to', false),
+            ],
+        );
+
+        $allContentTypes = $this->configurationService->isAllContentTypes();
+        $allowedContentTypes = $this->configurationService->getAllowedContentTypes();
+
+        if (!$allContentTypes && count($allowedContentTypes) === 0) {
+            $this->style->warning('No content types configured for scheduled visibility mechanism.');
+
+            return Command::FAILURE;
+        }
+
+        if (!$allContentTypes && count($allowedContentTypes) > 0) {
+            $criteria = new Criterion\LogicalAnd(
+                [
+                    $criteria,
+                    new Criterion\ContentTypeIdentifier($allowedContentTypes),
+                ],
+            );
+        }
+        $query->filter = $criteria;
+        $query->limit = 0;
+
+        $searchService = $this->repository->getSearchService();
```

Comment:
The problem with using `SearchService` to process the items is that we have no way to load the Content in the main translation only. So in a large multilingual repository this will not be performant, as loading Content fields is the most expensive operation. And the first requirement for this command is that it's performant, so it can be ran frequently.

Instead of doing this, implement fetching all Content IDs with a Doctrine DBAL. Then we can process the result in chunks and load the each Content item in the main translation only. We don't need to ensure that the required fields are not empty, that's an optimization we can add later on. Main Language ID can be found in the `ezcontentobject.initial_language_id` column.

You can get the Doctrine Connection object through `@ibexa.persistence.connection` service.

Reply:
added here e3ad2cf8056dc61107e7bc2244402ffa282c6b45, please check it out. I don't know if something like this is what you meant. I am also not sure if it should remain here as part of the command or it needs to extracted out.

Reply:
refactoring done here 830d104618d4fbe897df495e9fcccbd9b4565543 and here 50b5872844e461e98d07c118fcf5d9c12c9ae6ea